### PR TITLE
Fix github actions for agent container

### PR DESCRIPTION
**Pull request by olliewalsh**
The provision-ip-discovery-agent has been removed so drop from the
github actions. Add the operator-agent container instead. Also update
the agent container url in the CSV.

### CODE REVIEW

File: scripts/build_and_push_images.sh [hunk 1]
```diff
@@ -21,34 +21,54 @@ INDEX_IMG="$REPO/$OP_NAME-index:$VERSION"
...
+
 # Base operator image
 make manager
 make manifests
 make generate
-IMG=${IMG} make docker-build docker-push
+make IMG=${IMG} docker-build docker-push
+
+# Agent image
+make IMG=${AGENT_IMG} DOCKERFILE="Dockerfile.agent" docker-build docker-push
```
Comment (abays) [inline scripts/build_and_push_images.sh:37]: Just curious: we've previously placed env vars before the command as opposed to after it as args (such as in line 44).  This format works as well?

Comment (olliewalsh) [reply scripts/build_and_push_images.sh:37]: Yes. IIIRC the difference is that env vars will not override the var set in the Makefile.

File: scripts/build_and_push_images.sh [hunk 2]
```diff
@@ -21,34 +21,66 @@ INDEX_IMG="$REPO/$OP_NAME-index:$VERSION"
...
   base_image=$(echo $csv_image | cut -f 1 -d':')
   tag_image=$(echo $csv_image | cut -f 2 -d':')
   if [[ "$base_image:$tag_image" == "controller:latest" ]]; then
-    sed -e "s|$base_image:$tag_image|$IMG|g" -i bundle/manifests/osp-director-operator.clusterserviceversion.yaml
+    sed -e "s|$base_image:$tag_image|$IMG|g" -i ${CLUSTER_BUNDLE_FILE}
+  elif [[ "$base_image" == */"${AGENT_IMAGE}" ]]; then
+    sed -e "s|$base_image:$tag_image|$AGENT_IMG_WITH_DIGEST|g" -i "${CLUSTER_BUNDLE_FILE}"
+  elif [[ "$base_image" == */"${DOWNLOADER_IMAGE}" ]]; then
+    sed -e "s|$base_image:$tag_image|$DOWNLOADER_IMG_WITH_DIGEST|g" -i "${CLUSTER_BUNDLE_FILE}"
   else
```
[the remark this answers is not in the document]
Comment (olliewalsh) [reply scripts/build_and_push_images.sh:76]: That would give the digest for the images from the CSV, not the images we just built e.g would transform `registry.redhat.io/rhosp-rhel8-tech-preview/osp-director-downloader:1.0` to `registry.redhat.io/rhosp-rhel8-tech-preview/osp-director-downloader:sha...`

The */"${DOWNLOADER_IMAGE}" in the conditional will match the glob `*/osp-director-downloader` which is replaced with the quay url & digest for the build image.

[the remark this answers is not in the document]
Comment (dprince) [reply scripts/build_and_push_images.sh:76]: You are correct. So we'll need the extra changes here. All good then.

**Pull request merged**